From 1cd00878645a7690998cd479f5227774e0600bf0 Mon Sep 17 00:00:00 2001
From: Jacob Keller <jacob.e.keller@intel.com>
Date: Thu, 15 Nov 2012 01:10:37 +0000
Subject: [PATCH 689/858] ixgbe: add missing supported filters to get_ts_info

This patch updates the filters for ethtool's get_ts_info to return support for
all filters which can be supported by upscaling to ptp_v2_event. The intent
behind this change is due to reasoning that we do in fact support the
filters. (hwtstamp_ioctl returns success after setting the filter to the
upscaled version). In this way we can remain consistent over which filters are
supported via the get_ts_info ioctl and which filters are in practice actually
supported.

Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit aeb8264847c0194e42c8023cea89305c377f20e3)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_ethtool.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_ethtool.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_ethtool.c
index 21bc72e..1513b10 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_ethtool.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_ethtool.c
@@ -2709,6 +2709,14 @@ static int ixgbe_get_ts_info(struct net_device *dev,
    (1 << HWTSTAMP_FILTER_NONE) |
    (1 << HWTSTAMP_FILTER_PTP_V1_L4_SYNC) |
    (1 << HWTSTAMP_FILTER_PTP_V1_L4_DELAY_REQ) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L2_EVENT) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L4_EVENT) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_SYNC) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L2_SYNC) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L4_SYNC) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_DELAY_REQ) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L2_DELAY_REQ) |
+   (1 << HWTSTAMP_FILTER_PTP_V2_L4_DELAY_REQ) |
    (1 << HWTSTAMP_FILTER_PTP_V2_EVENT);
   break;
  default:
-- 
1.8.3.1